apiserver/userservice/userimpl.go

```diff
@@ -999,6 +1000,56 @@ func updateUser(request *msgs.UpdateUserRequest, cluster *crv1.Pgcluster) msgs.U
 		result.Password = password
 		sql = fmt.Sprintf("%s %s", sql,
 			fmt.Sprintf(sqlPasswordClause, util.SQLQuoteLiteral(hashedPassword)))
+
+		// Sync user to pgAdmin, if enabled
+		qr, err := pgadmin.GetPgAdminQueryRunner(apiserver.Clientset, apiserver.RESTConfig, cluster)
+		if err != nil {
+			log.Error(err)
+
+			result.Error = true
+			result.ErrorMessage = err.Error()
+
+			return result
+		} else if qr != nil {
+			// Get service details and prep connection metadata
+			service, svcFound, err := kubeapi.GetService(apiserver.Clientset, cluster.Name, cluster.Namespace)
+			if err != nil {
+				log.Error(err)
+
+				result.Error = true
+				result.ErrorMessage = err.Error()
+
+				return result
+			}
+
+			dbService := pgadmin.ServerEntry{}
+			if svcFound {
+				dbService = pgadmin.ServerEntryFromPgService(service, cluster.Name)
+			}
```

Comment:
I saw this logic elsewhere, too. It's okay for the service to not exist? If so, why use it at all?

Reply:
Fixed in upcoming commit

Reply:
addressed in upcoming commit